Runtimes/Core/CMakeLists.txt

```diff
@@ -181,7 +181,7 @@ add_compile_options(
 # We cannot selectively filter the linker warnings as we do not use the MSVC
 # frontned and `clang-cl` (and `clang`) currently do not support `/WX:nnnn`. As
 # a compromise, treat all linker warnings as errors.
-add_link_options($<$<PLATFORM_ID:Windows>:LINKER:/WX>)
+add_link_options($<$<PLATFORM_ID:Windows>:LINKER:/WX,/IGNORE:longsections>)
```

Comment:
Can we split this up and do this conditionally on debug information being enabled?

Reply:
AFAIK we don't have a good CMake flag we can use for that. Setting `DebugInfo` in `build.ps1` mostly changes the C/C++/Swift flags passed to the CMake invocation. We could parse these flags but that seems inherently brittle.

Reply:
On the contrary - I think that points out how to best handle this! `-DebugInfo` -> `-D CMAKE_EXE_LINKER_FLAGS="/IGNORE:longsections" -D CMAKE_SHARED_LINKER_FLAGS="/IGNORE:longsections"`.

Reply:
We are going to be hitting an issue with this, depending on the CMake version. Up to 3.29, this is going to have no effect. With 3.30-3.31, this is going to fail because `swiftc` does not know about `/IGNORE` flags. We could make it work with CMake 4.0+ with [CMP0181](https://cmake.org/cmake/help/latest/policy/CMP0181.html] set to `NEW` and prefixing `CMAKE_EXE_LINKER_FLAGS` and `CMAKE_SHARED_LINKER_FLAGS` with `LINKER:`.

But for Swift Ci, there is no way to pass the argument via these flags.